syntax = "proto3";
package file_transfer;

option go_package = "../proto/file_transfer";

service FileService {
    rpc Download(FileDownloadRequest) returns (stream FileChunk);
    rpc Upload(stream FileChunk) returns (FileResponse);
    rpc Delete(FileDeleteRequest) returns (FileResponse);
}

service EdgeFileService {
    rpc DownloadFromEdge(FileDownloadRequest) returns (stream FileChunk);
}

enum ErrorCodes {
    OK = 0;
    S3_ERROR = 9;
    CHUNK_ERROR = 10;
    INVALID_METADATA = 11;
    FILE_CREATE_ERROR = 12 ;
    FILE_NOT_FOUND_ERROR = 13;
    FILE_WRITE_ERROR = 15;
    FILE_READ_ERROR = 16;
    STREAM_CLOSE_ERROR = 17;
}

message FileDownloadRequest {
    string fileName = 1;
    string requestId = 2;
}

message FileChunk {
    bytes chunk = 1;
}

message FileResponse {
    string requestId = 1;
    bool success = 2;
}

message FileDeleteRequest {
    string fileName = 1;
    string requestId = 2;
}

